### PR TITLE
update UG

### LINKED ISSUE #215: [PE-D][Tester E] Borrow Command Abnormality
# Description 

When specifying the title of the book to borrow, it seems weird to me that the title specified has to be case-sensitive. I think it would be better if book titles are processed regardless of case-sensitivity as it would provide a simpler and easier way to type in the commands for fast typists. However, do let me know if this is an intended feature.

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: smolegz/ped#3

### LINKED ISSUE #216: [PE-D][Tester A] delbook command does not delete duplicated books
It only deletes one copy of duplicated book. Not sure if it is intended. 

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: Xuyan0518/ped#2

### LINKED ISSUE #218: [PE-D][Tester D] Email and phone are not unique identifiers
Good job on making the `name` a unique identifier for the contacts. However, I notice that `email` and `phone` are not unique identifiers. In real life, no two individuals can have the same email address or phone number, which should make them unique identifiers. I think making them unique identifiers would make it closer to real life representation. However, I do understand that it will make them less flexible in terms of the input.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: matthewken19/ped#2

### LINKED ISSUE #227: [PE-D][Tester E] Delete Book Command Design
# Description 

When deleting a book from the library, I believe that it would be more convenient for users to delete a book based on the book's registered index. Alternatively, it would be even better if users have the liberty to delete a book by either its title or index. This feature would enable greater flexibility for users.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: smolegz/ped#5

### LINKED ISSUE #230: [PE-D][Tester C]  Find command's UI picture not updated in the UG
It uses the old AB3's screenshots instead of the updated application.

-------------
Labels: `type.DocumentationBug` `severity.VeryLow`
original: jocelyn-soh/ped#7

### LINKED ISSUE #231: [PE-D][Tester E] UG Explanation for Donate Command
# Description 

In the UG's explanation for the `donate` command, it did not mention that donating a book would increase a user's merit score. A short statement could be added to let users know that this behavior exists, in the event that do not scroll down to the Note section.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: smolegz/ped#6

### LINKED ISSUE #239: [PE-D][Tester A] Successful deletion of person who has borrowed a book
this person has borrowed a book. But he can be successfully deleted and the book is also deleted together with him. 

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: Xuyan0518/ped#5

### LINKED ISSUE #241: [PE-D][Tester B] Lack of constraints in BOOKTITLE
This is quite a nit, really, but the lack of constraints for BOOKTITLE could open up possibilities of misuse and feature flaws.

1. Allowing duplicate book titles to be created, without other identifiers, is ambiguous as to which book is which book (feature-wise, as in real world use case). If a user borrows multiple books of the same title, which is being returned?

1. I stressed tested (not realistic use, really) it with an invisible symbol and was able to create a book title with invisible characters.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: dayn-1/ped#12

### LINKED ISSUE #262: [PE-D][Tester A] Ui bug
After successfully adding a book, the UI appears to be wrong

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: Xuyan0518/ped#9

### PR DESCRIPTION
**Pull request by Darkarche3**
closes #215 
closes #216 
closes #218 
closes #227 
closes #239 
closes #241 
closes #230 
closes #231 
closes #262

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -348,6 +349,13 @@ Furthermore, certain edits can cause the MyBookshelf to behave in unexpected way
 1. `add` command is used to record new user's personal information into the Contact List.
 1. `add` and `edit` Command **DOES NOT** support adding/editing merit score or borrowed book.
 1. `add`, `delete`, `edit`, `clear` and `find` commands are for managing users, while `addbook`, `delbook`, `borrow`, `donate` and `return` commands are for managing books.
+1. `borrow` command is cases-sensitive to allow books of similar titles to be differentiated.
+1. `delbook` deletes one book at a time to prevent accidental deletion of  all entries with the same book  title.
```
Comment (jinhanfromNUS) [inline docs/UserGuide.md:353]: Additional white space between "of" and "all"

File: docs/UserGuide.md [hunk 2]
```diff
@@ -196,7 +196,7 @@ Format: `find KEYWORD [MORE_KEYWORDS]`
 
 Examples:
 * `find alex david` returns `Alex Yeoh`, `David Li`<br>
-  ![result for 'find alex david'](images/findAlexDavidResult.png)
+  ![result for 'find alex david'](images/findJaneTanResult.png)
```
Comment (jinhanfromNUS) [inline docs/UserGuide.md:199]: I feel it is better to include the command as well. But no worries, I am going to fix it.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -348,6 +349,13 @@ Furthermore, certain edits can cause the MyBookshelf to behave in unexpected way
 1. `add` command is used to record new user's personal information into the Contact List.
 1. `add` and `edit` Command **DOES NOT** support adding/editing merit score or borrowed book.
 1. `add`, `delete`, `edit`, `clear` and `find` commands are for managing users, while `addbook`, `delbook`, `borrow`, `donate` and `return` commands are for managing books.
+1. `borrow` command is cases-sensitive to allow books of similar titles to be differentiated.
+1. `delbook` deletes one book at a time to prevent accidental deletion of  all entries with the same book  title.
+1. Email and phone numbers have not been made unique identifiers as there may be case scenarios where  two users might share the same email or phone number such as small child and a parent.
```
Comment (jinhanfromNUS) [inline docs/UserGuide.md:354]: Additional white space between "where" and "two"

File: docs/UserGuide.md [hunk 4]
```diff
@@ -360,6 +368,9 @@ Furthermore, certain edits can cause the MyBookshelf to behave in unexpected way
 
 ### Archiving data files
 
+### Introducing unique id for books
```
Comment (jinhanfromNUS) [inline docs/UserGuide.md:371]: I think this should be above ### Archiving data files, as this _Details coming soon_ should be the desription of ### Archiving data files

File: docs/UserGuide.md [hunk 5]
```diff
@@ -379,6 +390,7 @@ _Details coming soon..._
 5. Duplicated library users can be added with the same name but different capitalisation, due to our definition of duplicated people. Will be fixed in the future. Refer to the Developer Guide for more information.
 6. All parameters except `INDEX` and `THRESHOLD` do not have a limit to the number of characters. Refer to the Developer Guide for more information.
 7. The `clear` command only clears the library user data. The usage pertaining to this command will be made clearer with the implementation of future features. Refer to the **Future Features** section above for more information.
+8. The UI might not display text correctly if special characters or languages other than English is used in the input. For example, arabic is read from right to left but our application is only be able to read from left to right.
```
Comment (jinhanfromNUS) [inline docs/UserGuide.md:393]: Personally, I feel the 'a' from "arabic" can be capitalized.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -348,6 +349,13 @@ Furthermore, certain edits can cause the MyBookshelf to behave in unexpected way
 1. `add` command is used to record new user's personal information into the Contact List.
 1. `add` and `edit` Command **DOES NOT** support adding/editing merit score or borrowed book.
 1. `add`, `delete`, `edit`, `clear` and `find` commands are for managing users, while `addbook`, `delbook`, `borrow`, `donate` and `return` commands are for managing books.
+1. `borrow` command is cases-sensitive to allow books of similar titles to be differentiated.
+1. `delbook` deletes one book at a time to prevent accidental deletion of  all entries with the same book  title.
+1. Email and phone numbers have not been made unique identifiers as there may be case scenarios where  two users might share the same email or phone number such as small child and a parent.
+1. Book title is used instead of index as there is no urgent need for current functions like `borrow` or `addbook`. Furthermore, there is a plant to implement a findbook function in the future which will require use of book title. If booktitle is the same, include author name or other distinguishing features in brackets to differentiate book under `donate` and `addbook` commands.
```
Comment (jinhanfromNUS) [inline docs/UserGuide.md:355]: I am not sure, but do you mean "plan" instead of "plant"?

File: docs/UserGuide.md [hunk 7]
```diff
@@ -348,6 +349,13 @@ Furthermore, certain edits can cause the MyBookshelf to behave in unexpected way
...
+1. `delete` command will delete users along with any books borrowed. This is because community library manager cannot retrieve the books. If books are returned, community library manager can use `addbook` command to include the books back into the list.
+1. There is no constrain for book titles as it can be anything. For example, it could be in different languages like Chinese or Arabic.
```
Comment (jinhanfromNUS) [inline docs/UserGuide.md:357]: I think "constrain" should be spelt as "constraints"

I think there is a grammar error here for book titles and it.
I think it should be either
There **are** no constraints for book titles as **they** can be anything.
There **is** no constraint for book titles as **it** can be anything.

File: docs/UserGuide.md (hunk 2 above)
Comment (jinhanfromNUS) [inline docs/UserGuide.md:199]: I feel it is better to include the command as well. But no worries, I am going to fix it.

Comment (Darkarche3) [reply docs/UserGuide.md:199]: Showing the whole page might make it difficult for reader to understand what to focus on in the image as highlighted in another issue.

**Review by jinhanfromNUS — APPROVED** [review]

**Pull request merged**